Makefile

```diff
@@ -1,23 +1,45 @@
 test:
 	tox
 
-build:
-	rm -f dist/* && python3 -m pip install --upgrade build && python3 -m build .
+.PHONY: build
+build: setup_dependencies
+	rm -rf dist/* && poetry build && ./rename-wheel.sh
+
+install: setup_dependencies
+	poetry install --with dev
+
+.ONESHELL:
+setup_dependencies: check-poetry check-yarn
+	poetry config virtualenvs.create false
+	poetry self add "poetry-dynamic-versioning[plugin]"
+
+.SILENT:
+.PHONY: check-poetry
+check-poetry:
+	command -v poetry >/dev/null 2>&1 || { echo >&2 "Locust requires the poetry binary to be available in this shell to build the Python package.\nSee: https://docs.locust.io/en/stable/developing-locust.html#install-locust-for-development"; exit 1; }
```

Comment:
I may have found a nicer way to do this:
`which poetry > /dev/null ||  {...`

Reply:
Yeah I believe `which` is unix-specific - the windows equivalent would be `where` 

Come to think of it, I think `command` itself is POSIX compliant and I currently have no idea if this works on Windows either 😓 

Cross platform options likely boil down to:
- write a couple more lines (have cases for each platform)
- call out to the system's Python and have that check (we need Python anyway right)
- maybe just call the binary and catch the errors to the pipe anyway